### PR TITLE
Foederatis

### PR DESCRIPTION
**Pull request by loup99**
Worked on adding the planned foederati system to the mod: 

- An event chain with negotiations between Romans and "Barbarians"
- Both sides can initiate negotiations
- Failed negotiations can lead to war, while success can be profitable for both parties, provided that they manage to agree
- Possible outcomes is settlement within the empire and outside it, with promises or an immediate grant
- Promises are not necessarily respected, and can lead to considerable friction
- On death the foederati has a choice of leaving
- Added foederati modifiers to make them a greater threat than a normal neighbour
- A foederati may also be integrated into a vassal, at the risk of war
- New casus belli used in the event chain to impose Roman demands
- Romans securing a white peace or enforcing demands when defending against "Barbarians" can lead to negotiations

It is still work in progress and needs polishing/testing, as well as finished localisation

### CODE REVIEW

File: WTWSMS/events/WtWSMS_foederati_events.txt [hunk 1]
```diff
@@ -0,0 +1,2117 @@
...
+		name = WtWSMS_foederati.22.a
+		ai_chance = {
+			#Option A modifiers
+		}
+		custom_tooltip = {
+			text = WtWSMS_foederati.22.a.tooltip # Negotiations begin
+		}
+		event_target:roman_negotiator = {
+			character_event = { 
+				id = id = WtWSMS_foederati.1 # Romans accept or refuse to start negotiating with barbarian
```
Comment (IhateTrains) [inline WTWSMS/events/WtWSMS_foederati_events.txt]: Syntax error, caused parse failure

Comment (loup99) [reply WTWSMS/events/WtWSMS_foederati_events.txt]: Fixed in my local build.

Comment (IhateTrains) [reply WTWSMS/events/WtWSMS_foederati_events.txt]: Fixed by commit 5a21c6f :)

File: WTWSMS/events/WtWSMS_foederati_events.txt [hunk 2]
```diff
@@ -0,0 +1,2240 @@
...
+				factor = 0.5
+				NOT = { any_realm_province = { any_neighbor_province = { owner = { same_realm = FROM } } } }
+			}
+		}
+		custom_tooltip = {
+			text = WtWSMS_foederati.27.a.tooltip # War
+		}
+		if = {
+			limit = {
+				has_dlc = Jade_Dragon
```
Comment (IhateTrains) [inline WTWSMS/events/WtWSMS_foederati_events.txt]: "has_dlc = Jade Dragon" instead of "has_dlc = Jade_Dragon"

Comment (IhateTrains) [reply WTWSMS/events/WtWSMS_foederati_events.txt]: Fixed by commit b5bd11e

**Comment by IhateTrains** [comment]
I can't test now, but are the bugs we discovered fixed now?

**Pull request merged**